components/sidebar/sidebar_channel/sidebar_channel_link/sidebar_channel_link.tsx

```diff
@@ -132,10 +132,12 @@ export default class SidebarChannelLink extends React.PureComponent<Props, State
     removeTooltipLink = (): void => this.gmItemRef.current?.removeAttribute?.('aria-describedby');
 
     handleChannelClick = (event: React.MouseEvent<HTMLAnchorElement>): void => {
-        mark('SidebarChannelLink#click');
-        trackEvent('ui', 'ui_channel_selected_v2');
-
         this.handleSelectChannel(event);
+
+        setTimeout(() => {
```

Comment:
Why move these to a `setTimeout`? Would it perhaps make sense to use `requestAnimationFrame` instead?

Reply:
it was just a recommendation for separating concerns on the click handler, that wouldn't be execute immediately, so it woudln't delay the switch from channel to channel and team to team. The setTimeout makes it so that the initial event doesn't wait for the analytics.

Reply:
I think we might want to keep the `mark` call before `this.handleSelectChannel` then since that'll more accurately track how long it takes to select a channel. With the `mark` happening afterwards, we aren't measuring how long the call to `this.handleSelectChannel` takes which could be impacting the channel switch time depending on what it's doing